### ISSUE TITLE
Spec raises `invalidDid` on valid dids

### ISSUE THREAD

**Issue by aljones15** [w3c-ccg/did-method-key#60]
The following normative statement causes issues with the definition of `invalidDid` from [did-core](https://www.w3.org/TR/did-core/)

> Check the validity of the input identifier. The scheme MUST be the value did. The method MUST be the value key. The version MUST be convertible to a positive integer value. The multibaseValue MUST be a string and begin with the letter z. If any of these requirements fail, an invalidDid error MUST be raised.

The definition of `invalidDid` is:

> invalidDid
> The [DID](https://www.w3.org/TR/did-core/#dfn-decentralized-identifiers) supplied to the [DID resolution](https://www.w3.org/TR/did-core/#dfn-did-resolution) function does not conform to valid syntax. (See [3.1 DID Syntax](https://www.w3.org/TR/did-core/#did-syntax).)

The following statements should not raise `invalidDid`

- The method MUST be the value key. 
- The multibaseValue MUST be a string and begin with the letter z.
- The version MUST be convertible to a positive integer value.

The following is a valid did that breaks these rules: `did:sov:staging:PiEVD2uU2qKEQ5oxx1BJ6A`

I suggest adding a new error `invalidDidKey` for when something is a valid did, but not a valid `did:key`.

**Comment by peacekeeper** [comment]
I agree with most of what you're saying here, but I'm not sure if introducing too many DID method-specific errors (such as `invalidDidKeyVersion`) is very smart. This can become complicated for both resolvers and clients if you want to support multiple DID methods.

We could also make use of other metadata properties. E.g. a resolver could return a generic `invalidDid` or `notFound` error that every client understand, but in the metadata we could include additional information WHY it was invalid or not found, etc.
